keylime/verification/tpm_engine.py

```
@@ -739,12 +739,29 @@ def _process_results(self, failure: Any) -> None:
+                was_timed_out = self.attestation.agent.accept_attestations is False
+                had_failures = (self.attestation.agent.consecutive_attestation_failures or 0) > 0
+
+                self.attestation.agent.accept_attestations = True
 
-            # Re-enable attestations on successful attestation
-            # This allows PUSH mode agents to recover from timeout-induced FAIL status
-            self.attestation.agent.accept_attestations = True
+                if was_timed_out or not had_failures:
+                    self.attestation.agent.consecutive_attestation_failures = 0
+                elif config.getboolean("verifier", "auto_recover_attestation_failures", fallback=False):
+                    self.attestation.agent.consecutive_attestation_failures = 0
```

Ruff (0.15.12)

[warning] Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)